test/auth.js

```diff
@@ -36,8 +76,14 @@ describe('auth', function () {
 
     stubOpen = sinon.stub()
     stubOpen.throws('not stubbed')
+
+    tmpNetrc = require('tmp').fileSync().name
```

Comment:
this seems fine, but perhaps another time you could check out https://github.com/tschaub/mock-fs

I haven't used it myself, but it might make these kind of mocking things a bit simpler.